### DISCUSSION TITLE
header file

### DISCUSSION THREAD

**Discussion by khromenokroman** [strongswan/strongswan#1882] [Q&A]
tell me if you have header files for developers installed, if installed, where, thank you
https://docs.strongswan.org/docs/5.9/install/autoconf.html
it says here that in the prefix/lib directory and the prefix /usr, but when I want to do in vscode for example
#define <strongswan/charon.h>, then it tells me that the file is not found, what to do? tell me...

**Comment by tobiasbrunner** [comment]
If you are referring to the `--with-dev-headers` option, it takes a path where the headers will be installed. But it currently only installs the `libstrongswan` headers. Note that we don't provide any stable libraries or header files and you'll also require `config.h` after `configure` has been run. So it's usually easier to reference the build/source dir (see e.g. [this example of an out-of-tree plugin](https://github.com/tobiasbrunner/strongswan-redirect-plugin)).

> but when I want to do in vscode for example #define <strongswan/charon.h>, then it tells me that the file is not found, what to do? tell me...

Note that there is no `charon.h` file in our code base. What exactly are you trying to do with that?

**Reply by khromenokroman** [reply]
Thank you, I am charon.h I referred to for an example
